scipy/_lib/_util.py

```diff
@@ -129,6 +129,10 @@ def _lazywhere(cond, arrays, f, fillvalue=None, f2=None):
     """
     xp = array_namespace(cond, *arrays)
 
+    if is_dask_namespace(xp) or is_jax_namespace(xp):
+        # TODO: verify for jax
+        return xp.where(cond, f(arrays[0], arrays[1]), f2(arrays[0], arrays[1]) if not fillvalue else fillvalue)
```

Comment:
cf. https://github.com/scipy/scipy/pull/22070, cc @crusaderky.

Reply:
I think @mdhaber was also interested in changes to _lazywhere.

(Sorry for the slow ping!)

Reply:
This looks fine to me if you need something that works with Dask and JAX.

Reply:
-1 from me for this kind of specific hacks.
I would much rather have a compatibility layer lower down in the toolchain. `_lazywhere` is appropriate here (and IMHO, `_lazywhere` should live in array-api-extras).

Reply:
f2 can be None. IMHO _lazywhere should be moved to array-api-extras before such backend-specific hacks take place.

Reply:
I think I was try to hack around ``_lazywhere`` since IIRC @mdhaber mentioned ``_lazywhere`` didn't seem to be much of an optimization compared to just doing the regular where, and so might be removed.

Happy to wait for your jax/array-api-extra changes to land, though.
I still need to upstream some code to array-api-compat for this PR.